components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/obsrvability/handler/metrics/publisher/MetricHandler.java

```diff
@@ -0,0 +1,582 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.obsrvability.handler.metrics.publisher;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.AbstractExtendedSynapseHandler;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.API;
+import org.apache.synapse.rest.RESTUtils;
+import org.apache.synapse.transport.nhttp.NhttpConstants;
+import org.wso2.config.mapper.ConfigParser;
+import org.wso2.micro.integrator.core.internal.MicroIntegratorBaseConstants;
+import org.wso2.micro.integrator.obsrvability.handler.util.MetricConstants;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MetricHandler extends AbstractExtendedSynapseHandler {
+
+    public static final String METRIC_REPORTER = "metric_reporter";
+    public static final String INCREMENT_COUNT = "incrementCount";
+    public static final String GET_TIMER = "getTimer";
+
+    private Class loadedMetricClass;
+    private Method method;
+    private Object metricClassObject;
+
+    public void classLoader() {
```

Comment:
Change this method to getMetricReporter() and return the MetricReporter implementation from here